python/arcticdb/version_store/library.py

```diff
@@ -520,18 +521,20 @@ def write_batch(
 
         Returns
         -------
-        List[VersionedItem]
-            Structure containing metadata and version number of the written symbols in the store, in the
+        List[Union[VersionedItem, DataError]]
+            List of versioned items. The data attribute will be None for each versioned item.
+            i-th entry corresponds to i-th element of `payloads`. Each result correspond to
+            a structure containing metadata and version number of the written symbols in the store, in the
             same order as `payload`.
+            If a key error or any other internal exception is raised, a DataError object is returned, with symbol,
+            error_code, error_category, and exception_string properties.
 
         Raises
         ------
         ArcticDuplicateSymbolsInBatchException
             When duplicate symbols appear in payload.
         ArcticUnsupportedDataTypeException
```

Comment:
This still causes the entire method to raise, rather than returning a DataError?

Reply:
Same q for the other methods where this is still mentioned

Reply:
I guess so, not something I've looked into